pkg/controllers/gpu/controller.go

```diff
@@ -43,13 +43,18 @@ const (
 // +kubebuilder:rbac:groups=deviceplugin.intel.com,resources=gpudeviceplugins/status,verbs=get;update;patch
 
 // SetupReconciler creates a new reconciler for GpuDevicePlugin objects.
-func SetupReconciler(mgr ctrl.Manager) error {
-	c := &controller{scheme: mgr.GetScheme()}
-	return controllers.SetupWithManager(mgr, c, devicepluginv1.GroupVersion.String(), "GpuDevicePlugin", ownerKey)
+func SetupReconciler(mgr ctrl.Manager, namespace string, withWebhook bool) error {
+	c := &controller{scheme: mgr.GetScheme(), ns: namespace}
+	err := controllers.SetupWithManager(mgr, c, devicepluginv1.GroupVersion.String(), "GpuDevicePlugin", ownerKey)
+	if !withWebhook || err != nil {
+		return err
+	}
```

Comment:
nit: when choosing between increased cyclomatics and complex logical expressions I'd take the former. I guess it would be more readable with an additional `if` for `withWebhook` and without negation:
```go
if err := controllers.SetupWithManager(mgr, c, devicepluginv1.GroupVersion.String(), "GpuDevicePlugin", ownerKey); err != nil {
        return err
}

if withWebhook {
        return (&devicepluginv1.GpuDevicePlugin{}).SetupWebhookWithManager(mgr)
}

return nil
```

Reply:
good comment, changed, v3 pushed